uwuno/src/main/java/com/learning/uwuno/controllers/playerController.java

```diff
@@ -52,40 +36,82 @@ public String addPlayer(@RequestBody String json, @PathVariable String uid)
-            player.setName(parser.getValue("name"));
-        }
-        catch (NoSuchElementException e) {
-            throw new errorNotFound();
+            parser = new parser(json);
         }
         catch (JsonProcessingException e) {
             throw new internalServerError();
         }
+
+        // TODO: Change this to return a JSON Object, returning a POJO screws with JSON return
+        // Handle updating name
+        if (parser.exists("name")) {
+            try { // TODO: Check if name already exists and append number to it
+                containerService.updatePlayerName(uid, pid, parser.getValue("name"));
+                // Should we do it this way? Def good for debugging front end at least
+                return containerService.getPlayer(uid, pid);
+            }
+            catch (NoSuchElementException e) {
+                throw new errorNotFound();
+            }
+        }
+        // Handle drawing a card - remove a card from the deck and add to a player's cardList
+        else if (parser.exists("draw")) {
+            try {
+                int numToDraw = Integer.parseInt(parser.getValue("draw"));
+                return containerService.drawCards(uid, pid, numToDraw);
+            }
+            catch (NumberFormatException e) {
+                throw new badRequest();
+            }
+        }
+        // Handle playing a card - remove card from player and add to deck
+        // Will check for whether card is playable, however will throw if not
+        else if (parser.exists("cardType") &&
+                parser.exists("cardColor") &&
+                parser.exists("cardValue") &&
+                parser.exists("wildColor")) {
+            containerService.playCard(uid, pid, parser.getValue("cardType"),
+                                                parser.getValue("cardColor"),
+                                                parser.getValue("cardValue"),
+                                                parser.getValue("wildColor"));
+            return containerService.getPlayer(uid, pid).getCardList();
+        }
+        // Handle missing card information for playing
+        else if (parser.exists("cardType") ||
+                parser.exists("cardColor") ||
+                parser.exists("cardValue") ||
+                parser.exists("wildColor")) {
+            throw new badRequest(); // Separate for custom message later on?
+        }
+        else {
+            throw new badRequest();
+        }
     }
 
     // DELETES
-    @DeleteMapping(value = "rooms/{uid}/players")
-    public void deletePlayer(@RequestBody String json, @PathVariable String uid)
-            throws JsonMappingException, JsonProcessingException {
+    @DeleteMapping(value = "rooms/{uid}/players/{pid}")
+    public void deletePlayer(@RequestBody String json, @PathVariable String uid, @PathVariable String pid) {
         try {
-            parser parser = new parser(json);
```

Comment:
Moving as much logic out of controller as possible, controller is only meant to redirect the info to the relevant functions.